app/scripts/views/sign_up.js

```diff
@@ -314,6 +315,7 @@ function (Cocktail, p, BaseView, FormView, Template, AuthErrors, mailcheck,
 
           if (preVerifyToken && account.get('verified')) {
             self.logViewEvent('preverified.success');
+            self.onSignInSuccess(account);
```

Comment:
I was thinking more about this approach over lunch, and I think it can be simplified further. If [user.js](https://github.com/mozilla/fxa-content-server/blob/92425c5e0bf72779b9d51d447fcf47ddc2dad2df/app/scripts/models/user.js#L25) accepted a [notifications](https://github.com/mozilla/fxa-content-server/blob/92425c5e0bf72779b9d51d447fcf47ddc2dad2df/app/scripts/models/notifications.js) instance, user.js could [broadcast](https://github.com/mozilla/fxa-content-server/blob/92425c5e0bf72779b9d51d447fcf47ddc2dad2df/app/scripts/models/notifications.js#L42) the events over the notifications channel. Those events are automatically [propagated to the interTabCahnnel](https://github.com/mozilla/fxa-content-server/blob/92425c5e0bf72779b9d51d447fcf47ddc2dad2df/app/scripts/models/notifications.js#L44) and the mixins would not need to listen for events from the user model and act as a relay.

The new events could be added to the list of events that [`notifications` listens for on the interTabChannel and relays back to the app](https://github.com/mozilla/fxa-content-server/blob/92425c5e0bf72779b9d51d447fcf47ddc2dad2df/app/scripts/models/notifications.js#L15)

The signin/signout mixins could be reduced to only listening for the relayed events on `notifications` and redirecting as appropriate.